src/db.c

```diff
@@ -197,32 +193,46 @@ robj *lookupKeyWriteOrReply(client *c, robj *key, robj *reply) {
     return o;
 }
 
-/* Add the key to the DB.
+/* Add a key-value entry to the DB.
+ *
+ * A copy of 'key' is stored in the database. The caller must ensure the
+ * `key` is properly freed by calling decrRefcount(key).
  *
- * In this case a copy of `key` is copied in kvstore, the caller must ensure the `key` is properly freed.
+ * The value may (if its reference counter == 1) be reallocated and become
+ * invalid after a call to this function. The (possibly reallocated) value is
+ * stored in the database and the 'valref' pointer is updated to point to the
+ * new allocation.
  *
- * It's up to the caller to increment the reference
- * counter of the value if needed.
+ * The reference counter of the returned value is not incremented, so the caller
```

Comment:
Nit: the function doesn't return anything. Are you talking about `valref` here? Could it be more straightforward to write its name explicitly?